js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -106,51 +117,41 @@ const ValidatingController = modules.Controller.inherit((function() {
             const deferred = new Deferred();
             const completeList = [];
 
-            isFull = isFull || editingController.getEditMode() === EDIT_MODE_ROW;
+            const editMode = editingController.getEditMode();
+            isFull = isFull || editMode === EDIT_MODE_ROW;
 
             if(this._isValidationInProgress) {
                 return deferred.resolve(false).promise();
             }
 
             this._isValidationInProgress = true;
+            let fullValidating;
             if(isFull) {
+                fullValidating = new Deferred();
+                editingController.addDeferred(fullValidating);
```

Comment:
Why us the fullValidating variable needed? May be call

`editingController.addDeferred(deferred);`

The deferred variable is resolved after full validation.

Reply:
It makes sense, deferred should be sufficient.